invenio_vocabularies/contrib/awards/datastreams.py

```diff
@@ -172,20 +117,162 @@ def apply(self, stream_entry, **kwargs):
+
+    def read(self, item=None, *args, **kwargs):
+        """Reads the latest CORDIS Horizon Europe project zip file and yields an in-memory binary stream of it."""
+        if item:
+            raise NotImplementedError(
+                "CORDISProjectHTTPReader does not support being chained after another reader"
+            )
+
+        if self._origin == "HE":
+            file_url = "https://cordis.europa.eu/data/cordis-HORIZONprojects-xml.zip"
+        elif self._origin == "H2020":
+            file_url = "https://cordis.europa.eu/data/cordis-h2020projects-xml.zip"
+        elif self._origin == "FP7":
+            file_url = "https://cordis.europa.eu/data/cordis-fp7projects-xml.zip"
+        else:
+            raise ReaderError(
+                "The --origin option should be either 'HE' (for Horizon Europe) or 'H2020' (for Horizon 2020) or 'FP7'"
+            )
+
+        # Download the ZIP file and fully load the response bytes content in memory.
+        # The bytes content are then wrapped by a BytesIO to be file-like object (as required by `zipfile.ZipFile`).
+        # Using directly `file_resp.raw` is not possible since `zipfile.ZipFile` requires the file-like object to be seekable.
+        file_resp = requests.get(file_url)
+        file_resp.raise_for_status()
+        yield io.BytesIO(file_resp.content)
+
+
+class CORDISProjectTransformer(BaseTransformer):
+    """Transforms a CORDIS project record into an award record."""
+
+    def apply(self, stream_entry, **kwargs):
+        """Applies the transformation to the stream entry."""
+        record = stream_entry.entry
+        award = {}
+
+        # Here `id` is the project ID, which will be used to attach the update to the existing project.
+        award["id"] = f"00k4n6c32::{record['id']}"
+
+        categories = record["relations"]["categories"]["category"]
+        if isinstance(categories, dict):
+            categories = [categories]
+
+        award["subjects"] = [
+            {"id": f"euroscivoc:{category['code'].split('/')[-1]}"}
+            for category in categories
+            if category.get("@classification") == "euroSciVoc"
+        ]
+
+        organizations = record["relations"]["associations"]["organization"]
+        # Projects with a single organization are not wrapped in a list,
+        # so we do this here to be able to iterate over it.
+        organizations = (
+            organizations if isinstance(organizations, list) else [organizations]
+        )
+        award["organizations"] = []
+        for organization in organizations:
+            # Some organizations in FP7 projects do not have a "legalname" key,
+            # for instance the 14th participant in "SAGE" https://cordis.europa.eu/project/id/999902.
+            # In this case, fully skip the organization entry.
+            if "legalname" not in organization:
+                continue
+
+            organization_data = {
+                # TODO: Here the legal name is uppercase.
```

Comment:
I guess it doesn't matter because later we are gonna map the pic id to that in affiliations (once we have pic identifiers in affiliation) and then only store the ror ids here instead of pic